draft-ietf-quic-http.md

```diff
@@ -1202,9 +1202,17 @@ The following situations are examples of invalid PRIORITY frames:
 A PRIORITY frame with Empty bits not set to zero MAY be treated as a connection
 error of type HTTP_MALFORMED_FRAME.
 
-A PRIORITY frame that references a non-existent Push ID, a Placeholder ID
-greater than the server's limit, or a Stream ID the client is not yet permitted
-to open MUST be treated as a connection error of type HTTP_LIMIT_EXCEEDED.
+A PRIORITY frame that references a non-existent Push ID, or a Stream ID the
+client is not yet permitted to open MUST be treated as a connection error of
+type HTTP_LIMIT_EXCEEDED.
+
+A PRIORITY frame MAY reference a Placeholder ID that is equal to or greater than
+the server's advertised value of the `SETTINGS_NUM_PLACEHOLDERS` settings.  When
+receiving a PRIORITY frame with its prioritized element set to such a
+placeholder, the server SHOULD discard the frame without making any change to
+the priority tree.  When receiving a PRIORITY frame with its element dependency
+set to such a placeholder, the server SHOULD update the priority tree with the
+root of the tree being the dependency instead.
```

Comment:
With regards to #2690, should the default be the orphan or the root? Root makes a bit more sense, since this element is clearly something the client wants to explicitly prioritize. On the other hand, this is a bit dirty, since with the orphan placeholder, nothing else gets explicitly added to the root anymore.

Reply:
I agree that this is debatable.

The reason I've chosen "root" as the starting point of this PR is because it is the behavior defined in HTTP/2; I think we should try to retain similarity if possible (unless / until we decide to adopt a very different design). Note that orphan placeholder is still useful regardless of this; the placeholder minimizes the negative impact when PRIORITY frames on the control stream gets lost.

Hopefully, this point becomes a less of an issue once we recommend some sensible minimum for SETTINGS_NUM_PLACEHOLDERS; essentially guiding servers to support either none or at least 32 (or something alike).

Reply:
I feel like the orphan makes more sense -- just because the client is assigning *a* priority doesn't imply that it's assigning a *high* priority.  The implication would be that clients should build the priority tree with higher-priority placeholders having a lower placeholder ID.  At some point, the placeholder IDs cross the threshold of what the server supports, and everything in that tree region devolves to "after everything else."  That seems like a reasonable degradation path.

Reply:
> The implication would be that clients should build the priority tree with higher-priority placeholders having a lower placeholder ID.

This sounds like a good way of thinking. I agree that providing such a guidance is a better than having no guidance at all.

OTOH, That creates a dependency on #2690. @MikeBishop, should we wait for the merger of #2690 before updating the text?

Reply:
There's an [easy solution](https://github.com/quicwg/base-drafts/commit/053044d622bc6854474ee88d018a2ec88f7480dd) to that.

Reply:
I am impressed! 18c8303 changes the dependency to the orphan placeholder.